<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TestUi</class>
 <widget class="QWidget" name="TestUi">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>840</width>
    <height>608</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <widget class="QLabel" name="train_label">
   <property name="geometry">
    <rect>
     <x>320</x>
     <y>30</y>
     <width>81</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>20</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Train:</string>
   </property>
  </widget>
  <widget class="QComboBox" name="train_index">
   <property name="geometry">
    <rect>
     <x>410</x>
     <y>40</y>
     <width>51</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>18</pointsize>
    </font>
   </property>
   <property name="currentText">
    <string/>
   </property>
   <property name="currentIndex">
    <number>-1</number>
   </property>
  </widget>
  <widget class="QLabel" name="new_train_label">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>80</y>
     <width>61</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>New Train</string>
   </property>
  </widget>
  <widget class="QPushButton" name="new_train_button">
   <property name="geometry">
    <rect>
     <x>130</x>
     <y>80</y>
     <width>80</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>New Train</string>
   </property>
  </widget>
  <widget class="QLabel" name="commanded_speed_label_2">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>140</y>
     <width>111</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Commanded Speed</string>
   </property>
  </widget>
  <widget class="QPushButton" name="commanded_speed_button">
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>140</y>
     <width>80</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Set</string>
   </property>
  </widget>
  <widget class="QPushButton" name="actual_speed_button">
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>180</y>
     <width>80</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Set</string>
   </property>
  </widget>
  <widget class="QLabel" name="actual_speed_label">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>180</y>
     <width>111</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Actual Speed</string>
   </property>
  </widget>
  <widget class="QLabel" name="authority_label">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>220</y>
     <width>111</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Authority</string>
   </property>
  </widget>
  <widget class="QPushButton" name="authority_button">
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>220</y>
     <width>80</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Set</string>
   </property>
  </widget>
  <widget class="QLabel" name="emergency_brake_label">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>270</y>
     <width>101</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Emergency Brake</string>
   </property>
  </widget>
  <widget class="QPushButton" name="emergency_brake_button">
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>270</y>
     <width>80</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Toggle</string>
   </property>
  </widget>
  <widget class="QLabel" name="failure_mode_label">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>310</y>
     <width>101</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Failure Mode</string>
   </property>
  </widget>
  <widget class="QPushButton" name="failure_mode_button">
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>310</y>
     <width>80</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Activate</string>
   </property>
  </widget>
  <widget class="QFrame" name="cabin_info">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>70</y>
     <width>271</width>
     <height>261</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>1</number>
   </property>
   <widget class="QLabel" name="train_car_info_label">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>0</y>
      <width>141</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Train Car Info</string>
    </property>
   </widget>
   <widget class="QLabel" name="headlights_label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>50</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Headlights</string>
    </property>
   </widget>
   <widget class="QLabel" name="left_door_label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>130</y>
      <width>52</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Left Door</string>
    </property>
   </widget>
   <widget class="QLabel" name="right_door_label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>170</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Right Door</string>
    </property>
   </widget>
   <widget class="QLabel" name="cabin_lights_label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>90</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Cabin Lights</string>
    </property>
   </widget>
   <widget class="QLabel" name="headlight_status">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>50</y>
      <width>41</width>
      <height>21</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="cabin_lights_status">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>90</y>
      <width>41</width>
      <height>21</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="left_door_status">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>130</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="right_door_status">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>170</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="cabin_tem_label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>210</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Cabin Temp</string>
    </property>
   </widget>
   <widget class="QLabel" name="cabin_temp_value">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>210</y>
      <width>41</width>
      <height>21</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="w_label_2">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>210</y>
      <width>31</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>F</string>
    </property>
   </widget>
  </widget>
  <widget class="QLineEdit" name="commanded_speed">
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>140</y>
     <width>61</width>
     <height>22</height>
    </rect>
   </property>
   <property name="inputMethodHints">
    <set>Qt::ImhDigitsOnly</set>
   </property>
   <property name="inputMask">
    <string extracomment="Wrong"/>
   </property>
  </widget>
  <widget class="QLineEdit" name="authority">
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>220</y>
     <width>61</width>
     <height>22</height>
    </rect>
   </property>
   <property name="inputMethodHints">
    <set>Qt::ImhDigitsOnly</set>
   </property>
  </widget>
  <widget class="QLineEdit" name="actual_speed">
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>180</y>
     <width>61</width>
     <height>22</height>
    </rect>
   </property>
   <property name="inputMethodHints">
    <set>Qt::ImhDigitsOnly</set>
   </property>
  </widget>
  <widget class="QFrame" name="speed_frame">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>360</y>
     <width>411</width>
     <height>181</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <widget class="QLabel" name="actual_speed_label_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>50</y>
      <width>71</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Actual Speed</string>
    </property>
   </widget>
   <widget class="QLabel" name="commanded_speed_label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>90</y>
      <width>111</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Commanded Speed</string>
    </property>
   </widget>
   <widget class="QLabel" name="speed_info_label">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>0</y>
      <width>111</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Vital Info</string>
    </property>
   </widget>
   <widget class="QLabel" name="actual_speed_value">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>50</y>
      <width>41</width>
      <height>21</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="text">
     <string>234.2345</string>
    </property>
   </widget>
   <widget class="QLabel" name="commanded_speed_value">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>90</y>
      <width>41</width>
      <height>21</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string>53.3455</string>
    </property>
   </widget>
   <widget class="QLabel" name="mph_label">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>50</y>
      <width>31</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Mph</string>
    </property>
   </widget>
   <widget class="QLabel" name="mph_label_2">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>90</y>
      <width>31</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Mph</string>
    </property>
   </widget>
   <widget class="QLabel" name="service_brake_label">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>50</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Service Brake</string>
    </property>
   </widget>
   <widget class="QLabel" name="service_brake_status">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>50</y>
      <width>41</width>
      <height>21</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="emergency_brake_status_label">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>90</y>
      <width>101</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Emergency Brake</string>
    </property>
   </widget>
   <widget class="QLabel" name="emergency_brake_status">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>90</y>
      <width>41</width>
      <height>21</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="power_label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>130</y>
      <width>111</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Commanded Power</string>
    </property>
   </widget>
   <widget class="QLabel" name="power_value">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>130</y>
      <width>51</width>
      <height>21</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="w_label">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>130</y>
      <width>31</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>W</string>
    </property>
   </widget>
   <widget class="QLabel" name="authority_status_label">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>130</y>
      <width>101</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Authority</string>
    </property>
   </widget>
   <widget class="QLabel" name="authority_status">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>130</y>
      <width>41</width>
      <height>21</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
  </widget>
  <widget class="QLabel" name="announcement_label">
   <property name="geometry">
    <rect>
     <x>480</x>
     <y>420</y>
     <width>91</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Announcement</string>
   </property>
  </widget>
  <widget class="QLabel" name="announcement_status">
   <property name="geometry">
    <rect>
     <x>580</x>
     <y>420</y>
     <width>191</width>
     <height>21</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::Box</enum>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QLabel" name="beacon_label">
   <property name="geometry">
    <rect>
     <x>480</x>
     <y>470</y>
     <width>111</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Beacon</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="beacon_info">
   <property name="geometry">
    <rect>
     <x>590</x>
     <y>470</y>
     <width>171</width>
     <height>22</height>
    </rect>
   </property>
   <property name="inputMethodHints">
    <set>Qt::ImhDigitsOnly</set>
   </property>
   <property name="text">
    <string>Station: Penn Station; Door: Left</string>
   </property>
  </widget>
  <widget class="QPushButton" name="beacon_button">
   <property name="geometry">
    <rect>
     <x>610</x>
     <y>510</y>
     <width>80</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Send</string>
   </property>
  </widget>
  <widget class="QLabel" name="mph_label_3">
   <property name="geometry">
    <rect>
     <x>230</x>
     <y>140</y>
     <width>31</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Mph</string>
   </property>
  </widget>
  <widget class="QLabel" name="mph_label_4">
   <property name="geometry">
    <rect>
     <x>230</x>
     <y>180</y>
     <width>31</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Mph</string>
   </property>
  </widget>
  <widget class="QLabel" name="mph_label_5">
   <property name="geometry">
    <rect>
     <x>230</x>
     <y>220</y>
     <width>31</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Mph</string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
